text/3668-async-closures.md

```diff
@@ -563,6 +563,14 @@ let _ = || {
 
 Reusing the `async` keyword allows users to understand what an `async Fn() -> T` trait bound does by analogy, since they already should know that adding `async` to some `fn foo() -> T` makes it return an `impl Future<Output = T>` instead of the type `T`.
 
+### Wouldn't `F: AsyncFn() -> T` save more space for `async` trait bound modifiers?
+
+Some have argued that using `F: AsyncFn() -> T` rather than `F: async Fn() -> T` would better save space for whatever we might want the semantics to be if we were to later adopt generalized `async` trait bound modifiers.
+
+We don't think this is correct in a practical sense. If we were give meaning to and stabilize `AsyncFn()` trait bounds, and then later, due to work on generalized `async` trait bound modifiers, we were to give a different meaning to `async Fn()`, that would seem so undesirable that it's hard to imagine we would ever actually do it.  People would too strongly expect `AsyncFn()` and `async Fn()` to work in the same way.
```

Comment:
What about a future where `async Fn()` still has no meaning, but `async SomeOtherTrait` does, in a way that is not analogous to `AsyncFn()`?

Reply:
Then those meanings can coexist since by definition they don't overlap.

Reply:
I also feel like at this point it's not productive to debate against a "what if" that's quantified over a space of possibilities that's not really enumerable. I'm happy to say that any `async Trait` design that conflicts with at least the broad, user-facing observables of the definition of `async Fn` is not worth considering.